flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/physical/batch/DynamicPartitionPruningRule.java

```diff
@@ -0,0 +1,920 @@
+        return new BatchPhysicalDynamicFilteringTableSourceScan(
+                factScan.getCluster(),
+                factScan.getTraitSet(),
+                factScan.getHints(),
+                factScan.tableSourceTable(),
+                dynamicFilteringDataCollector);
+    }
+
+    private BatchPhysicalDynamicFilteringDataCollector createDynamicFilteringConnector(
+            RelNode dimSide, List<Integer> dynamicFilteringFieldIndices) {
+        final RelDataType outputType =
+                ((FlinkTypeFactory) dimSide.getCluster().getTypeFactory())
+                        .projectStructType(
+                                dimSide.getRowType(),
+                                dynamicFilteringFieldIndices.stream().mapToInt(i -> i).toArray());
+
+        return new BatchPhysicalDynamicFilteringDataCollector(
+                dimSide.getCluster(),
+                dimSide.getTraitSet(),
+                ignoreExchange(dimSide),
+                outputType,
+                dynamicFilteringFieldIndices.stream().mapToInt(i -> i).toArray());
+    }
+
+    private RelNode ignoreExchange(RelNode dimSide) {
+        if (dimSide instanceof Exchange) {
+            return dimSide.getInput(0);
+        } else {
+            return dimSide;
+        }
+    }
+
+    /** Simple dynamic filtering pattern with fact side in join right. */
+    protected static class DynamicPartitionPruningFactInRightRule
+            extends DynamicPartitionPruningRule {
+
+        public DynamicPartitionPruningFactInRightRule(RelRule.Config config) {
+            super(config);
+        }
+
+        /** Config. */
+        public interface Config extends RelRule.Config {
+            Config DEFAULT =
+                    EMPTY.withOperandSupplier(
+                                    b0 ->
+                                            b0.operand(BatchPhysicalJoinBase.class)
+                                                    .inputs(
+                                                            l ->
+                                                                    l.operand(
+                                                                                    BatchPhysicalRel
+                                                                                            .class)
+                                                                            .anyInputs(),
+                                                            r ->
+                                                                    r.operand(
+                                                                                    BatchPhysicalTableSourceScan
+                                                                                            .class)
+                                                                            .noInputs()))
+                            .withDescription("DynamicPartitionPruning:factInRight")
```

Comment:
just use the class name as the description

Reply:
> just use the class name as the description

Done!